fvtest/compilertest/tests/MetaDataTest.cpp

```diff
@@ -0,0 +1,159 @@
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "avl_api.h"
+#include "env/TRMemory.hpp"
+#include "runtime/CodeMetaDataManager.hpp"
+#include "runtime/CodeMetaDataPOD.hpp"
+#include <vector>
+
+
+/**
+ * The design of the CodeMetaData manager is such that it expects each code
+ * cache to be added to the CodeMetaDataManager. In order to avoid that
+ * coupling in this test, the below class allows allocating 'fake' homes for
+ * tree elements
+ */
+
+class TestEnhancedCodeMetaDataManager : public TR::CodeMetaDataManager {
+   public:
+   TestEnhancedCodeMetaDataManager() : TR::CodeMetaDataManager() {}
+
+   /**
+    * Reset the singleton on destruction.
+    */
+   ~TestEnhancedCodeMetaDataManager()
+      {
+      _codeMetaDataManager = NULL;
+      }
+
+   TR::MetaDataHashTable* addFalseCacheInfo(uintptr_t base, uintptr_t top) {
+
+      TR::MetaDataHashTable *newTable = allocateCodeMetaDataHash(base,top);
+
+      if (newTable)
+         {
+         avl_insert(_metaDataAVL, (J9AVLTreeNode *) newTable);
+         }
+
+      return newTable;
+   }
+};
+
+
+TEST(MetaDataTest, Construction) {
+   TR::CodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+}
+
+
+TEST(MetaDataTest, ExactInsertion) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+
+   TR::MethodMetaDataPOD data;
+   data.startPC = 10;
+   data.endPC   = 100;
+   data.name    = "test";
+
+   ASSERT_TRUE(manager.insertMetaData(&data));
+   ASSERT_TRUE(manager.containsMetaData(&data));
+
+   auto* metadata = manager.findMetaDataForPC(10);
+   ASSERT_TRUE(NULL != metadata);
+   ASSERT_TRUE(NULL !=  metadata->name);
+   ASSERT_STREQ("test", metadata->name);
+
+   ASSERT_TRUE(manager.removeMetaData(&data));
+}
+
+TEST(MetaDataTest, NonExactLookup) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+
+   TR::MethodMetaDataPOD data;
+   data.startPC = 10;
+   data.endPC   = 100;
+   data.name    = "test";
+
+   ASSERT_TRUE(manager.insertMetaData(&data));
+   ASSERT_TRUE(manager.containsMetaData(&data));
+
+   std::vector<int> test_points = {11, 25, 99};
+   for (auto i : test_points) { 
+      auto* metadata = manager.findMetaDataForPC(i);
+      ASSERT_TRUE(NULL != metadata) << "Didn't find pc" << i;
+      ASSERT_TRUE(NULL !=  metadata->name);
+      ASSERT_STREQ("test", metadata->name);
+   }
```

Comment:
It's not really clear to me what this test cases is actually testing. Could you add a description? Also, have you considered breaking these up into different test cases.

One last more advanced question (mostly just curios), is there a reason for picking this weak combination of equivalence partitioning and boundary-value analysis?